### PR TITLE
RELATED: RAIL-1815 - Make reference recordings more stable

### PR DESCRIPTION
**Pull request by lupko**
-  use stable stringify
-  replace projectId in URIs with workspace type
-  refresh all recordings
-  fix couple of test scenarios where URI had to be hardcoded (attr element coloring predicates)
-  fix unit/component tests where URI was snapshotted from ref workspace contents

---

Supported PR commands:

| Command         | Description            |
| --------------- | ---------------------- |
| `ok to test`    | Re-run standard checks |
| `extended test` | BackstopJS tests       |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

File: tools/mock-handling/src/recordings/common.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { IAnalyticalBackend } from "@gooddata/sdk-backend-spi";
 import fs = require("fs");
 import pick = require("lodash/pick");
 import isEmpty = require("lodash/isEmpty");
+import stringify from "json-stable-stringify";
```
Comment (no23reason) [inline tools/mock-handling/src/recordings/common.ts:6]: Detail: does this work? I mean this uses ESM import, while the other imports do not. If this works, we should consider using ESM import everywhere.

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/539/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2132/ : SUCCESS in 11m 58s

File: tools/mock-handling/src/recordings/common.ts (hunk 1 above)
Comment (lupko) [reply tools/mock-handling/src/recordings/common.ts:6]: yes it does (tested already on some recordings). imma going to change the imports everywhere

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/548/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2141/ : SUCCESS in 10m 54s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by lupko** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/551/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2144/ : FAILURE in 14m 24s

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/552/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/292/ : FAILURE in 16m 26s

**Comment by lupko** [comment]
extended test

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by lupko** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/554/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2145/ : SUCCESS in 11m 35s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/555/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/294/ : SUCCESS in 15m 04s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/295/ : FAILURE in 4m 34s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/551/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2146/ : SUCCESS in 10m 55s

**Pull request merged**